rskj-core/src/test/java/co/rsk/peg/BridgeSerializationUtilsTest.java

```diff
@@ -87,6 +87,94 @@ class BridgeSerializationUtilsTest {
+
+        BtcTransaction btcTx = new BtcTransaction(NETWORK_PARAMETERS);
+        btcTx.addInput(fundTx.getOutput(0));
+        btcTx.addInput(fundTx.getOutput(1));
+        btcTx.addOutput(Coin.COIN, OTHER_ADDRESS);
+
+        // Act
+        byte[] serializedBtcTransaction = BridgeSerializationUtils.serializeBtcTransaction(btcTx);
+        BtcTransaction deserializedBtcTransaction = BridgeSerializationUtils.deserializeBtcTransaction(serializedBtcTransaction, NETWORK_PARAMETERS, true);
+
+        // Assert
+        assertNotNull(serializedBtcTransaction);
+        assertNotNull(deserializedBtcTransaction);
+        assertEquals(btcTx, deserializedBtcTransaction);
+    }
+
+    @Test
+    void serializeAndDeserializeBtcTransaction_withValidDataAndWithoutInputs_shouldReturnEqualResults() {
+        // Arrange
+        BtcTransaction fundTx = new BtcTransaction(NETWORK_PARAMETERS);
+        fundTx.addOutput(Coin.FIFTY_COINS, ADDRESS);
+        fundTx.addOutput(Coin.FIFTY_COINS, OTHER_ADDRESS);
+
+        BtcTransaction btcTx = new BtcTransaction(NETWORK_PARAMETERS);
+        btcTx.addOutput(Coin.COIN, OTHER_ADDRESS);
+
+        // Act
+        byte[] serializedBtcTransaction = BridgeSerializationUtils.serializeBtcTransaction(btcTx);
+        BtcTransaction deserializedBtcTransaction = BridgeSerializationUtils.deserializeBtcTransaction(serializedBtcTransaction, NETWORK_PARAMETERS, false);
+
+        // Assert
+        assertNotNull(serializedBtcTransaction);
+        assertNotNull(deserializedBtcTransaction);
+        assertEquals(btcTx, deserializedBtcTransaction);
+    }
+
+    @ParameterizedTest
+    @NullSource
+    @EmptySource
+    void deserializeBtcTransaction_withInvalidData_shouldReturnNull(byte[] data) {
+        // Act
+        BtcTransaction deserializedTxWithInputs = BridgeSerializationUtils.deserializeBtcTransaction(data, NETWORK_PARAMETERS, true);
+        BtcTransaction deserializedTxWithoutInputs = BridgeSerializationUtils.deserializeBtcTransaction(data, NETWORK_PARAMETERS, false);
+
+        // Assert
+        assertNull(deserializedTxWithInputs);
+        assertNull(deserializedTxWithoutInputs);
+    }
+
+    @Test
+    void serializeAndDeserializeSvpFundTransaction_withValidData_shouldReturnEqualResults() {
+        // Arrange
+        BtcTransaction prevTx = new BtcTransaction(NETWORK_PARAMETERS);
+        prevTx.addOutput(Coin.FIFTY_COINS, ADDRESS);
+        prevTx.addOutput(Coin.FIFTY_COINS, OTHER_ADDRESS);
+
+        BtcTransaction svpFundTx = new BtcTransaction(NETWORK_PARAMETERS);
+        svpFundTx.addInput(prevTx.getOutput(0));
+        svpFundTx.addInput(prevTx.getOutput(1));
+        svpFundTx.addOutput(Coin.COIN, OTHER_ADDRESS);
+
+        // Act
+        byte[] serializedSvpFundTransaction = BridgeSerializationUtils.serializeBtcTransaction(svpFundTx);
+        BtcTransaction deserializedSvpFundTransaction = BridgeSerializationUtils.deserializeSvpFundTx(serializedSvpFundTransaction, NETWORK_PARAMETERS);
```

Comment:
Not sure if we really need this method that simply acts as a wrapper. Wdyt?

Reply:
thats one of the points of this refactor, having specific methods when deserializing txs we **know** they have inputs to avoid passing the boolean. 
We need to keep the whole `txHasInputs` thing since its part of bitcoin, but is not the cleanest thing to do 😄